src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -349,12 +331,11 @@ export class MeetingManager implements AudioVideoObserver {
   async listAndSelectDevices(): Promise<void> {
     await this.updateDeviceLists();
 
-    const hasAudioInput = this.audioInputDevices?.some(value => value.label) || false;
-    const hasAudioOutput = this.audioOutputDevices?.some(value => value.label) || false;
-    const hasVideoInput = this.videoInputDevices?.some(value => value.label) || false;
+    const hasAudioPermission = !!this.constraints.audio;
+    const hasVideoPermission = !!this.constraints.video;
```

Comment:
We can not check the existence of device `label` to determine whether or not to select devices. The design of device permission is different in browsers. Firefox and Safari exposes the device label and device id of camera/microphone even if they only obtain the permission of microphone/ camera.

To fit this difference, we need to store the `constraints` of `getUserMedia`, and do the check based on it.

Reply:
What happens when the user declines permissions for one or both constraint?

Reply:
> What happens when the user declines permissions for one or both constraint?

I'm not clear about the 'constraint'` you referred here. I assume you are talking about the device permission prompt.

For example, if users call `mm.join(DeviceLabels.Audio)` to join a meeting, SDK should let browser trigger the device permission prompt **ONLY for mic**. But in Firefox/Safari, there will be two permission prompts, the first one ask for mic permission, the second ask for camera permission, the latter will only appear if users allow the former.

So for your question, if users decline the first one, the second one won't appear, and there is no way to decline the both.

The problem here is, the second device permission prompt should not appear in anyway, this is a bug or incorrect design of PR #499, in which we allow users pass a `deviceLabels: DeviceLabels | DeviceLabelTrigger` to control the device permission logic in SDK.

This PR is to fix this bug or incorrect design of PR #499.

Reply:
In Firefox, `getUserMedia({audio: true, video: true})` produces a combined permissions prompt. The user can click 'Block', or the decline can be remembered, and the entire `gUM` call will be rejected.

We should not be triggering two prompts in any circumstance in Firefox or Chrome. I believe Safari does permission prompt coalescing, but haven't tested.

It's also possible to have no audio devices (typically on Linux) or no camera.

Why does a second prompt appear?

Reply:
> Why does a second prompt appear?

For example: 
1. users call `mm.join(DeviceLabels.Audio)`
2. `mm.setupDeviceLabelTrigger()` set `deviceLabelTrigger` to `getUserMedia({audio: true})`
3. `mm.updateDeviceLists()` which triggers the `deviceLabelTrigger`
    * First prompt appear
    * Here Firefox and Safari also get the device label, device Id of video devices as well
4. pass the check and set `hasVideoInput` to true [here](https://github.com/aws/amazon-chime-sdk-component-library-react/blob/d3a2570b7a90f98b0552c52ac23bc8c2a457b7ff/src/providers/MeetingProvider/MeetingManager.ts#L352-L354)
5. select the video device using its device Id
    * Second prompt appear (Firefox and Safari)
    * Cause in step 3 the device Id is exposed